From fb84406d975b6df497c94d66597c1c62cd578807 Mon Sep 17 00:00:00 2001
From: "Srivatsa S. Bhat (VMware)" <srivatsa@csail.mit.edu>
Date: Thu, 12 Jan 2023 15:26:04 -0800
Subject: [PATCH] linux-rt, ice: Fix build errors on kernel 6.1.y

Commit 4d213ee396a6 ("u64_stat: Remove the obsolete fetch_irq()
variants") in upstream PREEMPT_RT kernel deprecated
u64_stats_fetch_begin_irq() and u64_stats_fetch_retry_irq() and
converted all call-sites to use u64_stats_fetch_begin() and
u64_stats_fetch_retry() instead. Do the same in the ice driver.

Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---
 src/ice_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ice_main.c b/src/ice_main.c
index 7bdd417..88b11fc 100644
--- a/src/ice_main.c
+++ b/src/ice_main.c
@@ -8555,10 +8555,10 @@ ice_fetch_u64_stats_per_ring(struct ice_ring_stats *ring_stat, u64 *pkts,
 	if (!ring_stat)
 		return;
 	do {
-		start = u64_stats_fetch_begin_irq(&ring_stat->syncp);
+		start = u64_stats_fetch_begin(&ring_stat->syncp);
 		*pkts = ring_stat->stats.pkts;
 		*bytes = ring_stat->stats.bytes;
-	} while (u64_stats_fetch_retry_irq(&ring_stat->syncp, start));
+	} while (u64_stats_fetch_retry(&ring_stat->syncp, start));
 }

 /**
--
2.39.0

